### PR TITLE
feat(2d): add Path component

### PR DESCRIPTION
**Pull request by levirs565**
This will add `Path` component to render and animate SVG path.

This is part of [add Latex component pull request](https://github.com/motion-canvas/motion-canvas/pull/539).

`Path` component will inherit `Curve` component. This component can morph between two SVG path.

## Why Morphing Path is Necessary for Latex Tweening

When we render Latex, some character can render into different SVG path depending on its situtation.

For example: `(` and `)` character.

You can see these two math.

```
$$
\left( 1 \right)
$$
```

$$
\left( 1 \right)
$$

```
$$
\left( {1 \over 2} \right)
$$
```

$$
\left( {1 \over 2} \right)
$$

We can see that `(` and `)` in these two math look have two size. Path of `(` character in two math are actually different. When tweening latex between these math, we must morph path of `(` character.

### CODE REVIEW

**Comment by nick-1666** [comment]
Lets gooo!! one step closer to latex animation

File: packages/2d/src/curves/ArcSegment.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    ArcSegment.el.setAttribute(
+      'd',
+      `M ${this.startPoint.x} ${this.startPoint.y} A ${this.radius.x} ${this.radius.y} ${this.xAxisRotationDegree} ${this.largeArcFlag} ${this.sweepFlag} ${this.endPoint.x} ${this.endPoint.y}`,
+    );
+    this.length = ArcSegment.el.getTotalLength();
+
+    this.extremPoint = this.calculateExtremPoint();
+  }
+
+  // based on https://fridrich.blogspot.com/2011/06/bounding-box-of-svg-elliptical-arc.html
```
[the remark this answers is not in the document]
Comment (levirs565) [reply packages/2d/src/curves/ArcSegment.ts]: Actually same.

**Comment by levirs565** [comment]
I have fixed all issue.

Kute.js svgMorph actually cannot tween unclosed path well. Its also caanot tween between closed and unclosed path. It only can tween unclosed single line into closed shape. I create my own implementation to do that. I implement 2 method in 2 commit.

- First method is to immitate how [kute.js morph single line into circle](https://thednp.github.io/kute.js/svgMorph.html). When `createCurveProfileLerp` encounter twening between closed and unclosed path, it will force close unclosed path. However, this will make filled area gone. To prevent this, we must add extra polygon. This method is on [feat(2d): morph between closed and unclosed path](https://github.com/motion-canvas/motion-canvas/pull/700/commits/2d3827ffb96978ae77975076979ea349d812d55c)
- Second method is simpler than first. It will not force close unclosed path which means filled area not gone.  When interpolatin from unclosed to closed path, first all point is interpolated like unclosed path. When first point and last point is equal then path closed automatically. This will make different interpolation than first method but with simpler and performant code. This method is on [feat(2d): simple closed and unclosed path morphing](https://github.com/motion-canvas/motion-canvas/pull/700/commits/db2da325bde785759470da0a006bc286d3ae3d78).

In my opionion, we must use second method. Although the interpolation is different from KUTE.js, the code is much simpler and performant.

**Review by aarthificial — APPROVED** [review]

**Review by aarthificial — CHANGES_REQUESTED** [review]
The PR is generally good for merging but the documentation requires some updates.
Since #712, fiddles (` ```tsx editor `) are required to correctly import all modules they use.

**Comment by levirs565** [comment]
I have fixed documentation error.

**Review by aarthificial — APPROVED** [review]

**Comment by aarthificial** [comment]
Awesome! Thanks for the PR and apologies for the delays.

**Comment by ksassnowski** [comment]
Thanks a ton @levirs565! Whenever you're ready, please go ahead and open the next PR. We really appreciate your work on this!

**Pull request merged**